tests/foreman/api/test_docker.py

```diff
@@ -0,0 +1,662 @@
+        altering the other products.
+
+        @Feature: Docker
+
+        """
+        content_type = u'docker'
+        upstream_name = u'busybox'
+        product_ids = []
+        repository_ids = []
+
+        for _ in range(0, randint(1, 5)):
+            product_ids.append(
+                entities.Product(organization=self.org_id).create()['id'])
+
+        for product_id in product_ids:
+            name = gen_string('utf8', 15)
+            repository = _create_repository(product_id, name, upstream_name)
+            repository_id = repository['id']
+            real_attrs = entities.Repository(id=repository_id).read_json()
+            self.assertEqual(real_attrs['name'], name)
+            self.assertEqual(real_attrs['docker_upstream_name'], upstream_name)
+            self.assertEqual(real_attrs['content_type'], content_type)
+            repository_ids.append(repository_id)
+
+        # Delete a ramdom repository
+        shuffle(repository_ids)
+        repository_id = repository_ids.pop()
+        entities.Repository(id=repository_id).delete()
+        with self.assertRaises(HTTPError):
+            entities.Repository(id=repository_id).read_json()
+
+        # Check if others repositories are not touched
+        for repository_id in repository_ids:
+            repository = entities.Repository(id=repository_id).read_json()
+            self.assertIn(repository['product']['id'], product_ids)
+
+
+@ddt
+class DockerContentViewTestCase(APITestCase):
+    """Tests specific to using ``Docker`` repositories with Content Views."""
+
+    @classmethod
+    def setUpClass(cls):  # noqa
+        """Create an organization which can be re-used in tests."""
+        cls.org_id = entities.Organization().create()['id']
+
+    @run_only_on('sat')
+    @data(
+        gen_string('alpha', 15),
+        gen_string('alphanumeric', 15),
+        gen_string('numeric', 15),
+        gen_string('latin1', 15),
+        gen_string('utf8', 15),
+        gen_string('html', 15),
+    )
+    def test_add_docker_repo_to_content_view(self, name):
+        """@Test: Add one Docker-type repository to a non-composite content view
+
+        @Assert: A repository is created with a Docker repository and the
+        product is added to a non-composite content view
+
+        @Feature: Docker
+
+        """
+        upstream_name = u'busybox'
+        content_type = u'docker'
+        prod_id = entities.Product(organization=self.org_id).create()['id']
+
+        repository = _create_repository(prod_id, name, upstream_name)
+        repo_id = repository['id']
+        real_attrs = entities.Repository(id=repo_id).read_json()
+        self.assertEqual(real_attrs['name'], name)
+        self.assertEqual(real_attrs['docker_upstream_name'], upstream_name)
+        self.assertEqual(real_attrs['content_type'], content_type)
+
+        # Create content view and associate docker repo
+        content_view = _create_content_view(self.org_id, composite=False)
```

Comment:
`composite` defaults to `False` in the method definition, so you can drop the second parameter here altogether. The purpose of `_create_content_view` is to make the code more compact anyway, right? :)